config/generic-recore-a7.cfg

```diff
@@ -0,0 +1,284 @@
+[tmc2209 stepper_z2]
+uart_pin: ar100:PD3
+uart_address: 0
+run_current: 0.500
+hold_current: 0.500
+stealthchop_threshold: 999999
+
+
+[stepper_x]
+step_pin: ar100:PL4
+dir_pin: ar100:PE8
+endstop_pin: !ar100:PH4
+rotation_distance: 8
+microsteps: 16
+position_endstop: 0
+position_max: 200
+
+[stepper_y]
+step_pin: ar100:PL5
+dir_pin: ar100:PE9
+endstop_pin: !ar100:PH5
+rotation_distance: 8
+microsteps: 16
+position_endstop: 0
+position_max: 200
+
+[stepper_z]
+step_pin: ar100:PL6
+dir_pin: ar100:PE10
+endstop_pin: !ar100:PH6
+rotation_distance: 8
+microsteps: 16
+position_endstop: 0
+position_max: 200
+
+[stepper_z1]
+step_pin: ar100:PL10
+dir_pin: ar100:PE14
+rotation_distance: 8
+microsteps: 16
+
+[stepper_z2]
+step_pin: ar100:PL11
+dir_pin: ar100:PE15
+rotation_distance: 8
+microsteps: 16
+
+[extruder]
+step_pin: ar100:PL7
+dir_pin: ar100:PE11
+heater_pin: PA8
+sensor_type: EPCOS 100K B57560G104F
+sensor_pin: PA0
+rotation_distance: 33.500
+microsteps: 16
+nozzle_diameter: 0.400
+filament_diameter: 1.75
+control: pid
+pid_Kp: 22.2
+pid_Ki: 1.08
+pid_Kd: 114
+min_extrude_temp: 30
+min_temp: -272
+max_temp: 300
+
+[extruder1]
+step_pin: ar100:PL8
+dir_pin: ar100:PE12
+heater_pin: PA9
+sensor_pin: PA1
+sensor_type: EPCOS 100K B57560G104F
+rotation_distance: 33.500
+microsteps: 16
+nozzle_diameter: 0.400
+filament_diameter: 1.75
+control: pid
+pid_Kp: 22.2
+pid_Ki: 1.08
+pid_Kd: 114
+min_extrude_temp: 30
+min_temp: -272
+max_temp: 300
+
+[extruder2]
+step_pin: ar100:PL9
+dir_pin: ar100:PE13
+heater_pin: PA10
+sensor_pin: PA2
+sensor_type: EPCOS 100K B57560G104F
+rotation_distance: 33.500
+microsteps: 16
+nozzle_diameter: 0.400
+filament_diameter: 1.75
+control: pid
+pid_Kp: 22.2
+pid_Ki: 1.08
+pid_Kd: 114
+min_extrude_temp: 30
+min_temp: -272
+max_temp: 300
+
+[heater_bed]
+heater_pin: PA11
+sensor_pin: PA3
+sensor_type: EPCOS 100K B57560G104F
+control: watermark
+min_temp: -272
+max_temp: 300
+
+[fan]
+pin: PF0
+
+[output_pin fan1]
+pin: PB1
+
+[output_pin fan2]
+pin: PB5
+
+[temperature_fan board]
+pin: PB4
+min_temp: 0
+max_temp: 100
+sensor_type: EPCOS 100K B57560G104F
+sensor_pin: PA6
+control: watermark
+gcode_id: Board
+target_temp: 50
+
+[temperature_sensor cold_junction]
+sensor_type: EPCOS 100K B57560G104F
+sensor_pin: PA7
+gcode_id: CJ
+
+[adc_temperature v]
+temperature1: 0.35
+voltage1: 0
+temperature2: 36.65
+voltage2: 3.3
+
+[temperature_sensor voltage]
+adc_voltage: 3.3
+sensor_pin: PA4
+sensor_type: v
+gcode_id: Voltage
+
+[adc_temperature current]
+temperature1: 0
+voltage1: 0
+temperature2: 165
+voltage2: 3.3
+
+[temperature_sensor current]
+adc_voltage: 3.3
+sensor_pin: PA5
+sensor_type: current
+max_temp: 20
+gcode_id: Current
+
+[gcode_button over_current_alarm]
+pin: !ar100:PF6
+press_gcode: M112
+
+[adc_temperature fan_current]
+temperature1: 0
+voltage1: 0
+temperature2: 33
+voltage2: 3.3
+
+[temperature_sensor fan_current]
+adc_voltage: 3.3
+sensor_pin: PB0
+sensor_type: fan_current
+max_temp: 2.0
+gcode_id: FanCurrent
```

Comment:
Out of curiosity, is there a reason for these gcode_id fields?  They only change the output of the `M105` command which, as far as I know, none of the modern interfaces use.